### ISSUE TITLE
Episode 236, when register with twitter or facebook

### ISSUE THREAD

**Issue by TeuF** [ryanb/railscasts-episodes#3]
Hi,

I have a problem: if I register using twitter or facebook instance, the password field stay blank. Then if the user try to login using email + password

I get the following error:

BCrypt::Errors::InvalidHash in Devise/sessionsController#create
invalid hash

How can I prevent this to happen ?

Thanks a lot.

**Comment by jpmckinney** [comment]
This is because the encrypted_password field is empty. Seeding the password with a random string when people create users via Twitter/Facebook/etc. will avoid this problem.

**Comment by ryanjm** [comment]
If you do that, won't it stop them from being able to add a password later?

Ideally, it would return saying that email/password is incorrect, right?

**Comment by jpmckinney** [comment]
Yes, it would, but if they are logging in with Facebook, they don't need a password on your site (right?).

**Comment by ryanjm** [comment]
It's true, I just happen to let the user be able to login multiple ways. Idea is that authentication != identification, so why not let the user be able to login through multiple means?

**Comment by jpmckinney** [comment]
In that case, I would require users to set a password on your site when they login with Facebook for the first time.
